File: 1894-merge-strings-alternately/1894-merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
       
        if (word1.isEmpty()){
            return word2; 
        }
        else if (word2.isEmpty()) {
            return word1;
        }

        String word = "";
        if (word1.length() >= word2.length()){
            for (int i = 0; i < word1.length(); i++) {
                if (i >= word2.length()) {
                    word += word1.charAt(i);
                }
                else {
                    word += word1.charAt(i);
                    word += word2.charAt(i);
                }
            }
        }
        else {
            for (int i = 0; i < word2.length(); i++){
                if (i >= word1.length()) {
                    word += word2.charAt(i);
                }
                else {
                    word += word1.charAt(i);
                    word += word2.charAt(i);
                }
            }
        }
        return word;
    }
}

File: 1894-merge-strings-alternately/1894-merge-strings-alternately.py
class Solution(object):
    def mergeAlternately(self, word1, word2):
        """
        :type word1: str
        :type word2: str
        :rtype: str
        """
        i = 0
        j = 0
        word = ''
        while (i < len(word1) and j < len(word2)):
            word += word1[i]
            word += word2[i]
            i+=1
            j+=1
        
        if (i < len(word1)):
            word += word1[i: len(word1)]

        elif (j < len(word2)):
            word += word2[i: len(word2)]

        return word 

        
